src/main/java/core/basesyntax/strategy/operation/SupplyOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.Fruit;
+
+public class SupplyOperation implements OperationHandler {
+    private StorageDao storageDao = new StorageDaoImpl();
+
+    @Override
+    public void operationType(String fruit, int amount) {
+        if (storageDao.getFruit(fruit) != null) {
+            int newQuantity = storageDao.getFruit(fruit).getQuantity() + amount;
+            storageDao.update(fruit, newQuantity);
```

Comment:
You're calling `storageDao.getFruit(fruit)` twice, which could be inefficient if the underlying implementation is complex. Consider storing the result in a local variable and reusing it.